doi/src/main/java/ca/nrc/cadc/doi/PostAction.java

```diff
@@ -463,11 +474,78 @@ private void performDoiAction() throws Exception {
             log.debug("redirectUrl: " + redirectUrl);
             syncOutput.setHeader("Location", redirectUrl);
             syncOutput.setCode(303);
+        } else if (doiAction.equals(SEARCH_ACTION)) {
+            if (doiStatusSearchFilter == null || callersNumericId == null) {
+                getStatusList(getOwnedDOIList());
+            } else {
+                getStatusList(getFilteredDOIList(doiStatusSearchFilter));
+            }
         } else {
             throw new UnsupportedOperationException("DOI action not implemented: " + doiAction);
         }
     }
-    
+
+    private List<Node> getFilteredDOIList(DoiStatusSearchFilter doiStatusSearchFilter) throws Exception {
+        List<Node> filteredNodes = new ArrayList<>();
+        ContainerNode doiRootNode = vospaceDoiClient.getContainerNode("");
```

Comment:
Again this should use getDOIContainer() in the base action because not finding the parent DOI node should result in an exception.